### ISSUE TITLE
False positive `open_curly_linter` / `pipe_continuation_linter` in magrittr pipelines

### ISSUE THREAD

**Issue by AshesITR** [r-lib/lintr#1028]
There is no way to make this piece of code lint-free without rewriting:

```r
letters %>%
{ # - open_curly_linter
  tibble(
    lo = .,
    hi = toupper(.)
  )
} %>%
  mutate(row_id = row_number())

letters %>% { # - pipe_continuation_linter
  tibble(
    lo = .,
    hi = toupper(.)
  )
} %>%
  mutate(row_id = row_number())
```

**Comment by MichaelChirico** [comment]
Slight clarification: the trailing comment is not part of the issue:

```r
letters %>%
{
  tibble(
    lo = .,
    hi = toupper(.)
  )
} %>%
  mutate(row_id = row_number())
```

also lints.

We should probably prefer to align to what `styler` does in this case (though it may be hard to predict if `styler` will change behavior in the future unless the action is tied to a specific style rule):

```r
styler::style_text("
letters %>%
{
  tibble(
    lo = .,
    hi = toupper(.)
  )
} %>%
  mutate(row_id = row_number())
")
```

gives

```r
letters %>%
  {
    tibble(
      lo = .,
      hi = toupper(.)
    )
  } %>%
  mutate(row_id = row_number())
```

Starting from the second option gives the same:

```r
styler::style_text("
letters %>% {
  tibble(
    lo = .,
    hi = toupper(.)
  )
} %>%
  mutate(row_id = row_number())
")
```

```r
letters %>%
  {
    tibble(
      lo = .,
      hi = toupper(.)
    )
  } %>%
  mutate(row_id = row_number())
```

**Comment by MichaelChirico** [comment]
Very similar to #487 (maybe the same?)

**Comment by AshesITR** [comment]
Worth considering separately for the to-be `brace_linter`.

**Comment by bersbersbers** [comment]
> Worth considering separately for the to-be `brace_linter`.

This is still an issue with `brace_linter` as of 36a18e00b9694da5b1c579891da1a19c3a7a250a, see also

https://github.com/r-lib/lintr/blob/afcbdb3fa7ee7b75681e273c968cec4701529372/R/brace_linter.R#L53-L54

**Comment by bersbersbers** [comment]
> we should add an exclusion if the preceding token is a magrittr pipe.

Adding a comma and an opening parenthesis to that exclusion would also fix #487 if I am not mistaken (but I may be!).
